pkg/controller/workload/job/job_controller.go

```diff
@@ -170,21 +183,35 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 
 	// 3. handle a finished job
 	if jobFinished {
-		added := false
-		wl.Status.Conditions, added = appendFinishedConditionIfNotExists(wl.Status.Conditions, jobFinishedCond)
-		if !added {
+		if workload.InCondition(wl, kueue.WorkloadFinished) {
 			return ctrl.Result{}, nil
 		}
+		condition := generateFinishedCondition(jobFinishedCond)
+		apimeta.SetStatusCondition(&wl.Status.Conditions, condition)
 		err := r.client.Status().Update(ctx, wl)
 		if err != nil {
 			log.Error(err, "Updating workload status")
 		}
 		return ctrl.Result{}, err
 	}
 
+	// handle a job when waitForPodsReady is enabled
+	if r.waitForPodsReady {
+		log.V(5).Info("Handling a job when waitForPodsReady is enabled")
+		condition := generatePodsReadyCondition(&job)
```

Comment:
I don't see a practical scenario when it matters. Maybe if the request to set admission=nil, but the request to set the suspend=true fails (and is not executed as the job gets quickly re-admitted), but then it seems to make sense to react to the change of the `suspend` field.

Reply:
The only scenario where it could matter is when the workload is immediately admitted again and we didn't have time to update the PodsReady condition to false.

But that could still happen in any case, because we can't update the status and spec at the same time. Unless we can in a mutating webhook?

Reply:
We can't reset the condition in a mutating webhook either. To be on the safer side, maybe we can set the condition to false as soon as we see `admission=nil`.